components/permalink_view/actions.ts

```diff
@@ -1,27 +1,27 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {getChannel, getChannelMember, selectChannel, joinChannel, getChannelStats} from 'mattermost-redux/actions/channels';
-import {getPostThread} from 'mattermost-redux/actions/posts';
-import {getCurrentTeam, getTeam} from 'mattermost-redux/selectors/entities/teams';
-import {getCurrentUser, getUser} from 'mattermost-redux/selectors/entities/users';
-
-import {loadChannelsForCurrentUser} from 'actions/channel_actions.jsx';
-import {loadNewDMIfNeeded, loadNewGMIfNeeded} from 'actions/user_actions.jsx';
-import {browserHistory} from 'utils/browser_history';
-import {joinPrivateChannelPrompt} from 'utils/channel_utils';
-import {ActionTypes, Constants, ErrorPageTypes} from 'utils/constants';
-import {getUserIdFromChannelId, isSystemAdmin} from 'utils/utils';
+import { getChannel, getChannelMember, selectChannel, joinChannel, getChannelStats } from 'mattermost-redux/actions/channels';
+import { getPostThread } from 'mattermost-redux/actions/posts';
+import { getCurrentTeam, getTeam } from 'mattermost-redux/selectors/entities/teams';
+import { getCurrentUser, getUser } from 'mattermost-redux/selectors/entities/users';
+
+import { loadChannelsForCurrentUser } from 'actions/channel_actions';
+import { loadNewDMIfNeeded, loadNewGMIfNeeded } from 'actions/user_actions';
+import { browserHistory } from 'utils/browser_history';
+import { joinPrivateChannelPrompt } from 'utils/channel_utils';
+import { ActionTypes, Constants, ErrorPageTypes } from 'utils/constants';
+import { getUserIdFromChannelId, isSystemAdmin } from 'utils/utils';
 
 let privateChannelJoinPromptVisible = false;
 
-export function focusPost(postId, returnTo = '', currentUserId) {
-    return async (dispatch, getState) => {
+export function focusPost(postId: string, returnTo?: string, currentUserId?: string): Promise<void> {
+    return async (dispatch: any, getState: any) => {
```

Comment:
Can you help avoid using `any` here? There are `DispatchFunc` and `GetStateFunc` definitions available too.